packages/next-auth/src/core/lib/oauth/callback.ts

```diff
@@ -161,25 +159,19 @@ export interface GetProfileParams {
   logger: LoggerInstance
 }
 
-export interface GetProfileResult {
-  // @ts-expect-error
-  profile: ReturnType<OAuthConfig["profile"]> | null
-  account: Omit<Account, "userId"> | null
-  OAuthProfile: Profile
-}
-
 /** Returns profile, raw profile and auth provider details */
 async function getProfile({
   profile: OAuthProfile,
   tokens,
   provider,
   logger,
-}: GetProfileParams): Promise<GetProfileResult> {
+}: GetProfileParams) {
   try {
     logger.debug("PROFILE_DATA", { OAuthProfile })
-    // @ts-expect-error
     const profile = await provider.profile(OAuthProfile, tokens)
     profile.email = profile.email?.toLowerCase()
+    if (!profile.id) throw new TypeError("Profile id is missing")
```

Comment:
We could log which Provider it's here:
```
Profile ID is missing in Twitter OAuth profile response
```
Thinking in case the user has configured multiple OAuth providers, it wouldn't be obvious 🤔